In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats

%precision 3
%matplotlib inline

In [2]:
df=pd.read_csv('ch4_scores400.csv')
scores=np.array(df['score'])

- 모평균(p_mean)
- 모분산(p_var)

In [5]:
p_mean=np.mean(scores)
p_var=np.var(scores)

p_mean, p_var

(69.53, 206.6691)

In [6]:
np.random.seed(0)
n=20
sample=np.random.choice(scores,n)

np.random.seed(1111)
n_samples=10000
samples=np.random.choice(scores, (n_samples, n))

## 점추정
- 표본평균 (s_mean)
- 불편분산(s^2) (u_var)

In [7]:
s_mean=np.mean(sample)
s_mean

70.4

In [8]:
u_var=np.var(sample, ddof=1)
u_var

158.25263157894742

## 구간추정
- 표본o, 모분산 알고있을 경우 : 모평균의 95% 신뢰구간 => z분포
- 표본o, 모분산 모름 : 모분산의 95% 신뢰구간 => 카이제곱 분포
- 표본o, 모분산 모름 : 모평균의 95% 신뢰구간 => t분포

(1) 표본o, 모분산 알고있을 경우 / 모평균 신뢰구간 (z분포)

In [11]:
rv=stats.norm()
#0.025 : 0.05/2
#0.975 : 1-0.05/2
lcl=s_mean-rv.isf(0.025)*np.sqrt(p_var/n)
ucl=s_mean-rv.isf(0.975)*np.sqrt(p_var/n)

lcl, ucl

(64.09956018535247, 76.70043981464754)

모평균이 69.53점 이었으므로 신뢰구간 내에 포함되어있음을 알 수 있음

(2) 표본o, 모분산 모름 / 모분산 신뢰구간 (카이제곱 분포)

In [12]:
rv=stats.chi2(df=n-1)
#0.025 : 0.05/2
#0.975 : 1-0.05/2
lcl=(n-1)*u_var/rv.isf(0.025)
ucl=(n-1)*u_var/rv.isf(0.975)

lcl, ucl

(91.52471947132238, 337.5955129124592)

모분산은 206.669였으므로 구간 내에 포함되어있다는 것을 확인

(3) 표본o, 모분산 모름 / 모평균 신뢰구간 (t분포)

In [14]:
rv=stats.t(df=n-1)
#0.025 : 0.05/2
#0.975 : 1-0.05/2
lcl=s_mean-rv.isf(0.025)*np.sqrt(u_var/n)
ucl=s_mean-rv.isf(0.975)*np.sqrt(u_var/n)

lcl, ucl

(64.51244886603887, 76.28755113396114)

모평균의 값은 59.53점 이었으므로 신뢰구간 내에 포함되어있음을 알 수 있음

- 예제 1: 어느 회사에서 생산하는 제품의 모분산을 추리하기 위하여, 표본 10개를 뽑았더니 표본분산은 6.42가 나왔다고 한다. 이때 제품의 모분산에 대한 90%의 신뢰구간을 구하시오.

In [18]:
# 표본o, 모분산x / 모분산 신뢰구간

In [19]:
n=10
u_var=6.42
rv=stats.chi2(df=n-1)

lcl=(n-1)*u_var/rv.isf(0.05)
ucl=(n-1)*u_var/rv.isf(0.95)

lcl, ucl

(3.415099975321245, 17.37685387744868)

- 예제 2: 향수 20개 추출, 지속시간의 분산 90**2, 향수의 분산 90% 신뢰구간을 구하라

In [22]:
# 표본o, 모분산x / 모분산 신뢰구간

In [28]:
n=20
u_var=90**2
rv=stats.chi2(df=n-1)

lcl=(n-1)*u_var/rv.isf(0.025)
ucl=(n-1)*u_var/rv.isf(0.975)

lcl, ucl

(4684.59968293086, 17279.482984311377)